### DISCUSSION TITLE
解决部分资源文件无法序列化/还原问题

### DISCUSSION THREAD

**Discussion by TOUCHERS** [icsharpcode/ILSpy#2633] [Ideas]
反编译后的项目，部分资源文件仍无法序列化或者还原（图片）？
尝试解决方法：使用System.Resources.Extensions.DeserializingResourceReader读取资源文件(需要添加nuget包)
位置：ILSpy-master\ICSharpCode.Decompiler\CSharp\ProjectDecompiler\WholeProjectDecompiler.cs
修改WriteResourceToFile函数，
原
```
foreach (var entry in new ResourcesFile(entryStream))
{
	writer.AddResource(entry.Key, entry.Value);
}
```
改为
```
foreach (System.Collections.DictionaryEntry entry in new System.Resources.Extensions.DeserializingResourceReader(entryStream))
{
	writer.AddResource(entry.Key.ToString(), entry.Value);
}
```
由此可以解决

**Comment by dgrunwald** [comment]
Deserialization leads to a security vulnerability: #1196
